static/js/public/snap-details/channelMap.js

```diff
@@ -1,294 +1,418 @@
-      let iframe = document.querySelector('.js-snap-open-frame');
+  slideToInstructions(clickEl) {
+    // Add content to the right slide area
+    this.writeInstallInstructions(clickEl.dataset.channel);
 
-      if (iframe) {
-        iframe.parentNode.removeChild(iframe);
-      }
+    const slides = clickEl.closest('.p-channel-map__slides');
+    slides.classList.add('show-right');
+    slides.classList.remove('show-left');
+  }
 
-      iframe = document.createElement('iframe');
-      iframe.className = 'js-snap-open-frame';
-      iframe.style.position = 'absolute';
-      iframe.style.top = '-9999px';
-      iframe.style.left = '-9999px';
-      iframe.src = `snap://${name}`;
-      document.body.appendChild(iframe);
-
-      if (typeof ga !== 'undefined') {
-        // The first attempt should be counted towards the 'intent'
-        let label = 'Snap install intent';
-        let value = `${name}`;
-
-        // Subsequent attempts should still be tracked, but not as 'intent'
-        if (attempt > 1) {
-          label = 'Snap install click';
-          value += ` - click ${attempt}`;
-        }
+  writeInstallInstructions(channel) {
+    let paramString = '';
 
-        ga('gtm1.send', {
-          hitType: 'event',
-          eventCategory: 'Snap details',
-          eventAction: 'Click view in desktop store button',
-          eventLabel: label,
-          eventValue: value
-        });
+    if (channel.indexOf('latest') === 0) {
+      if (channel.indexOf('stable') !== 7) {
+        paramString = `--${channel.split('/')[1]}`;
       }
-
-      attempt += 1;
+    } else {
+      paramString = `--channel=${channel}`;
     }
-  });
-}
-
-export default function initChannelMap(el, packageName, channelMapData) {
-  initOpenSnapButtons();
 
-  const channelMapEl = document.querySelector(el);
-  const channelOverlayEl = document.querySelector('.p-channel-map-overlay');
+    const template = this.INSTALL_TEMPLATE
+      .split('${channel}')
+      .join(channel)
+      .split('${paramString}')
+      .join(paramString);
+    const holder = document.querySelector('[data-js="channel-map-install-details"]');
 
-  initTabs(channelMapEl);
+    holder.innerHTML = template;
+  }
 
-  let closeTimeout;
+  writeTable(el, data) {
+    let cache;
+    const tbody = data.map((row, i) => {
+      const isSameTrack = (cache && row[0] === cache);
+      let rowClass = [];
 
-  // init open/hide buttons
-  const openChannelMap = (event) => {
-    const openButton = event.target.closest('.js-open-channel-map');
+      if (i === 0) {
+        rowClass.push('is-highlighted');
+      }
 
-    if (openButton) {
-      // open tab based on button click (or install tab by default)
-      const openTabName = openButton.getAttribute('aria-controls') || 'channel-map-tab-install';
+      if (isSameTrack) {
+        rowClass.push('no-border');
+      }
 
-      // clear hiding animation if it's still running
-      clearTimeout(closeTimeout);
```

Comment:
On the other note - it's a bit pity that while we started to use HTML template in other place (for install window) we still keep a bunch of quite complicated HTML in the JS.

Reply:
I've moved it :+1: